lib/Ansi/Ansi.mli

```diff
@@ -0,0 +1,182 @@
+(** This module defines the escape sequence data type.
+
+    Every value of this type is one styling ([`Bold],
+    [`Reverse], [`Foreground] {!Color.red}) or the composition
+    of two of them ([`Bold]{{!(&)}[&]}[`Reverse]), which can be chained
+    ([`Dim] {{!(&)}[&]} [`Italic] {{!(&)}[&]} [`Blink] {{!(&)}[&]} [`Background] {!Color.magenta}).
+
+    {[
+    open Ansifmt
+
+    let chesnay = `Italic
+    let rocquencout = Ansi.(`Bold & `Reverse & `Foreground Color.red)
+
+    (* compose to infinity! *)
+    let inria = Ansi.(chesnay & rocquencourt)
+    ]}
+
+    {i [chesnay], [rocquencourt] and [inria] all have the same
+    type {!t}.}
+
+    In a composition chain, if two stylings are overlapping
+    (for example, [`Foreground Color.blue & `Foreground Color.red]),
+    the rightmost takes precedence.
+    *)
+
+(** Represents an ANSI SGR escape sequence. *)
+type t =
+  [ `Bold
+  | `Dim
+  | `Italic
+  | `Underline
+  | `Blink
+  | `Reverse
+  | `Foreground of Color.t
+  | `Background of Color.t
+  | `Composed of t * t
+  ]
+
+(** [compose left right] combines the [left] and right
+    sequences into one.
+
+    See also: {{!(&)}[( & )]} *)
+val compose : t -> t -> t
+
+(** [to_attributes ansi] produces an attribute record of the
+    [ansi] escape sequence. *)
+val to_attributes : t -> Attributes.t
+
+(** [to_cancelling_attributes ansi] produces an attribute
+    record of the escape sequence that would cancel the effects
+    of [ansi]. *)
+val to_cancelling_attributes : t -> Attributes.t
```

Comment:
Does it make sense to (now or later) add a function to take a string with attributes and cancelling attributes and validate if any of the cancelling attributes have no effect for that string, assuming no previous state changes?

e.g. if I have ```
let start = `Composed (`Bold, `Blink) in
let cancelling = `Composed (`Reverse, `Bold)
```
then a hypothetical `has_irrelevant_cancelling_attributes` should return `[\`Reverse]` because it is the only one that was not turned on in the first place?